src/components/body/audio/sound-context.js

```diff
@@ -0,0 +1,192 @@
+import { Howl } from 'howler'
+import PropTypes from 'prop-types'
+import React, { PureComponent } from 'react'
+import memoizeOne from 'memoize-one'
+
+const updateInterval = 200 // ms
+
+const createEmptyFunction = funcName => () => {
+  console.warn(
+    'An empty method that should be overwritten in `ControlContext.Provider` is invoked. The mehtod name is',
+    funcName
+  )
+}
+
+export const TimeContext = React.createContext({
+  current: 0,
+  duration: 10, // any number > 0 is okay
+})
+
+export const ControlsContext = React.createContext({
+  play: createEmptyFunction('play'),
+  stop: createEmptyFunction('stop'),
+  pause: createEmptyFunction('pause'),
+  toggleMute: createEmptyFunction('toggleMute'),
+  setCurrent: createEmptyFunction('setCurrent'),
+})
+
+export const StatusContext = React.createContext({
+  isPlaying: false,
+  isMute: false,
+})
+
+export class SoundProvider extends PureComponent {
+  static propTypes = {
+    src: PropTypes.string.isRequired,
+    children: PropTypes.node,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      // The `current`, `duration`, `isPlaying`, and `isMute` in state are just the snapshot of status in `this._sound`.
+      // In the other words, setting state won't affect `this._sound`.
+      current: 0,
+      duration: 0,
+      isPlaying: false,
+      isMute: false,
+    }
+    this.stop = this.stop.bind(this)
+    this.play = this.play.bind(this)
+    this.pause = this.pause.bind(this)
+    this.setCurrent = this.setCurrent.bind(this)
+    this.toggleMute = this.toggleMute.bind(this)
+    this._createStatusProps = memoizeOne((isMute, isPlaying) => ({
+      isMute,
+      isPlaying,
+    }))
+    this._createTimeProps = memoizeOne((current, duration) => ({
+      current,
+      duration,
+    }))
+    this._createControlProps = memoizeOne(
+      (play, pause, stop, setCurrent, setDuration, toggleMute) => ({
+        play,
+        pause,
+        stop,
+        setCurrent,
+        setDuration,
+        toggleMute,
+      })
+    )
```

Comment:
I use `memoizeOne` to prevent creating new object for the prop of `Context.Provider`

Reply:
Why do we need to prevent creating new object? For avoid re-rendering on Consumer?
Since `current` is changing all the time, will `memoizeOne` cause performance issues?

Reply:
If we want to prevent from creating new object due to the same values(`current`, `isPlaying` ... etc), I recommend that we could use [immutable](https://github.com/immutable-js/immutable-js)

Reply:
The `this.state.current` may change rapidly (at least every 200~300ms), when other props stay no changed

Reply:
> Why do we need to prevent creating new object? For avoiding re-rendering on Consumer?
> Since `current` is changing all the time, will `memoizeOne` cause performance issues?

Sorry for wrong understanding about `memoizeOne`.
Since `memoizeOne` only memoize last one object creation, not all object creations, it won't have performance issues here.